dspy/teleprompt/mipro_optimizer_v2.py

```diff
@@ -400,7 +402,7 @@ def _bootstrap_fewshot_examples(
                 teacher_settings=self.teacher_settings,
                 seed=seed,
                 metric_threshold=self.metric_threshold,
-                rng=self.rng,
+                parallel_bootstrapping=self.parallel_bootstrapping,
```

Comment:
interesting, i gotta check what the semantics are of defaulting to self.attribute.....

Reply:
btw why False by default?

Reply:
Partially it's in case it breaks something I didn't test for and partially because the printing is currently ugly, but I can change the logic to support num_threads

Reply:
How do we want to handle `num_threads`?  Theoretically BootstrapFewshotWithRandomSearch will call Evaluate which means if num_threads=6 one interpretation is allowing 6*6=36 simultaneous threads, but this is likely not what a user expects.  We could allow two different variables for num_threads (num_threads_eval, num_threads_bootstrap), or we could set each to the sqrt of num_threads (with max bootstrap threads being limited to num_candidates+3)

Reply:
Oh, hmm you're mentioning BootstrapFewshotWithRandomSearch? I thought we're parallelizing across BootstrapFewShot. Is that wrong?

Reply:
Currently I am parallelizing **calls to** BootstrapFewShot rather than BootstrapFewShot itself.  I see what you are saying though and can refactor to that instead.  MIPRO has a util function that basically runs BootstrapFewshotWithRandomSearch just without the final evaluation, so my comment earlier about threads within threads is not a concern if we pick to parallelize EITHER **calls to** BootstrapFewShot or BootstrapFewShot.  Parallelizing BootstrapFewShot has the benefit that it can work for both RandomSearch and MIPRO out of the box.

Reply:
>Parallelizing BootstrapFewShot has the benefit that it can work for both RandomSearch and MIPRO out of the box.

Yes, that's the better one.